finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4DispatcherBase.scala

```diff
@@ -50,10 +53,9 @@ trait Netty4DispatcherBase[SendMsg <: Message, RecvMsg <: Message] {
     log.debug("[%s S:%d] initialized stream", prefix, id)
     val _ = stream.onReset.respond {
       case Return(_) =>
-        // Free and clear.
-        if (streams.replace(id, open, StreamClosed)) {
-          log.debug("[%s S:%d] stream closed", prefix, id)
-        }
+        closedId.updateAndGet(new IntUnaryOperator { def applyAsInt(i: Int) = if (id > i) id else i })
```

Comment:
Unless you want to pull in scala-java8-compat, there probably isn't. Java 8 lets you specify these operators as lambdas. 

https://github.com/scala/scala-java8-compat/blob/master/src/test/scala/scala/compat/java8/FunctionConvertersTest.scala#L388